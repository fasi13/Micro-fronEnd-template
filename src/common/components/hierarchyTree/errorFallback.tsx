import React from 'react';
import { FallbackProps } from 'react-error-boundary';

export const ErrorFallback = ({
	error,
	resetErrorBoundary,
}: FallbackProps): JSX.Element => (
	<div role="alert" className="text-center text-white">
		<p>Something went wrong with loading your Application Hierarchy Tree:</p>
		<pre>{error.message}</pre>
		<button type="button" onClick={resetErrorBoundary}>
			Try again
		</button>
	</div>
);
