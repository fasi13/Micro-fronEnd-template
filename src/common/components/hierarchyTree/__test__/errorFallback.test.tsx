import { render } from '@testing-library/react';
import React from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import { ErrorFallback } from '../errorFallback';

let Bomb: React.FunctionComponent<{ shouldThrow: boolean }>;
let App: React.FunctionComponent<{ shouldThrow: boolean }>;
describe('ErrorBoundary and Fallback', () => {
	beforeEach(() => {
		Bomb = ({ shouldThrow }) => {
			if (shouldThrow) throw new Error('dummy error I made up');
			return <div>BOMB HERE</div>;
			return null;
		};

		App = (props: { shouldThrow: boolean }) => {
			const { shouldThrow } = props;
			return (
				<div>
					Testing errorboundary
					<div>
						<ErrorBoundary FallbackComponent={ErrorFallback}>
							<Bomb shouldThrow={shouldThrow} />
						</ErrorBoundary>
					</div>
				</div>
			);
		};
	});
	it('renders error boundary and reports error if thrown', () => {
		const { getByRole } = render(<App shouldThrow />);
		expect(getByRole('alert')).toBeInTheDocument();
		expect(getByRole('alert')).toMatchSnapshot();
	});

	it('dose not render error boundary if no error is thrown', () => {
		const { queryByRole } = render(<App shouldThrow={false} />);
		expect(queryByRole('alert')).not.toBeInTheDocument();
	});
});
