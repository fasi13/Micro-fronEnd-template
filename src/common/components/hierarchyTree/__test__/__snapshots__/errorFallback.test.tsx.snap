// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary and Fallback renders error boundary and reports error if thrown 1`] = `
<div
  class="text-center text-white"
  role="alert"
>
  <p>
    Something went wrong with loading your Application Hierarchy Tree:
  </p>
  <pre>
    dummy error I made up
  </pre>
  <button
    type="button"
  >
    Try again
  </button>
</div>
`;
